### PR TITLE
clusterctl: fix install and upgrade to v0.3.15

### LINKED ISSUE #4454: Homebrew formula builds the wrong path
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]
The [Homebrew formula](https://github.com/Homebrew/homebrew-core/blob/990506a976e22cc77d57d424ef18dbf9de077697/Formula/clusterctl.rb#L17-L19) I've created in #2389 seems wrong:

```rb
  def install
    system "go", "build", *std_go_args, "-o", bin/"clusterctl"
  end
```

It should build the `cmd/clusterapi/main.go`, or call `make clusterctl`. I'm right?

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]
 
/cc @fabriziopandini

### PR DESCRIPTION
**Pull request by ludusrusso**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Closes https://github.com/kubernetes-sigs/cluster-api/issues/4454

### CODE REVIEW

File: Formula/clusterctl.rb [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
+  sha256 "5f5b70b2a5469bb1c70079a186e1dfec6542fc870f1f95db236082ac3b70a955"
   license "Apache-2.0"
 
-  bottle do
-    sha256 cellar: :any_skip_relocation, arm64_big_sur: "5a7c6535dc4831815d681d2d786eab329a5fc9b4b687427540a98191c7393663"
-    sha256 cellar: :any_skip_relocation, big_sur:       "af40e87e380e35e3647891635efc12c60494339c65c4c553f0341fe3a319be0f"
-    sha256 cellar: :any_skip_relocation, catalina:      "2bf9c2331956328df8e8cb0ea0037ef8786e9201cb1cc0aa372ce295aaeee481"
-    sha256 cellar: :any_skip_relocation, mojave:        "4a2c710b1166aeed3a5216ee1747204c2daac183c6a4c1e72e97dbba8a97a5c4"
-  end
-
```
Comment (carlocab) [inline Formula/clusterctl.rb:14]: Please don't touch the bottle block.

File: Formula/clusterctl.rb [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
   depends_on "go" => :build
+  depends_on "make" => :build
```
Comment (carlocab) [inline Formula/clusterctl.rb:9]:
```suggestion
```
You only need this if you need `gmake`, but you don't seem to be using it.

File: Formula/clusterctl.rb [hunk 3]
```diff
@@ -1,25 +1,20 @@
...
 
   def install
-    system "go", "build", *std_go_args, "-o", bin/"clusterctl"
+    system "make", "clusterctl"
+    prefix.install Dir["#{buildpath}/bin"]
```
Comment (carlocab) [inline Formula/clusterctl.rb:13]:
```suggestion
    prefix.install "bin"
```
Does this not work?

**Review by carlocab — COMMENTED** [review]
Thanks, @ludusrusso. Just a few suggestions. When you make these changes, please squash them with your first commit and use the commit message
```
clusterctl: fix install and upgrade to v0.3.15
```

**Comment by ludusrusso** [comment]
@carlocab thanks for help!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**